Makefile

```diff
@@ -363,15 +363,18 @@ endif
 #	make bench-daily TO=/path/to/file.json
 .PHONY: bench-daily
 bench-daily:
-	go test github.com/pingcap/tidb/pkg/session -run TestBenchDaily -bench Ignore --outfile bench_daily.json
+	go test github.com/pingcap/tidb/pkg/distsql -run TestBenchDaily -bench Ignore --outfile bench_daily.json
 	go test github.com/pingcap/tidb/pkg/executor -run TestBenchDaily -bench Ignore --outfile bench_daily.json
 	go test github.com/pingcap/tidb/pkg/executor/test/splittest -run TestBenchDaily -bench Ignore --outfile bench_daily.json
-	go test github.com/pingcap/tidb/pkg/tablecodec -run TestBenchDaily -bench Ignore --outfile bench_daily.json
 	go test github.com/pingcap/tidb/pkg/expression -run TestBenchDaily -bench Ignore --outfile bench_daily.json
-	go test github.com/pingcap/tidb/pkg/util/rowcodec -run TestBenchDaily -bench Ignore --outfile bench_daily.json
-	go test github.com/pingcap/tidb/pkg/util/codec -run TestBenchDaily -bench Ignore --outfile bench_daily.json
-	go test github.com/pingcap/tidb/pkg/distsql -run TestBenchDaily -bench Ignore --outfile bench_daily.json
+	go test github.com/pingcap/tidb/pkg/planner/core/tests/partition -run '^[^T]' -bench . -benchmem \
+		| awk 'BEGIN{printf("[")} /^Benchmark.*ns\/op/ {if (found>0) {printf(",")}; sub(/-[0-9]*/,"",$$1); printf("{\"Name\":\"%s\",\"NsPerOps\":%d,\"AllocsPerOp\":%d,\"BytesPerOp\":%d}",$$1,$$3,$$7,$$5); found++ } END{printf("]")}' \
```

Comment:
I found the generate json file can not pass json pretty format, so this maybe not quite stable.
If you do not call `benchdaily.Run()` directly, maybe you can wrap around this? The utilities are all there and simple enough.

https://github.com/pingcap/tidb/blob/81da4f8811f2fc150ed176d7ef3bcde3ce112d9f/pkg/util/benchdaily/bench_daily.go#L77-L78

Reply:
You can expose `benchmarkResultToJSON`

Reply:
I updated the Makefile with gobenchdata + gojq to extract and format the daily_bench.json file instead.
The issue with `testing.Benchmark(t)` is that it runs a full 'main' benchmark, and sums all the 'sub-benchmarks', so no handy way of getting each sub-benchmark numbers.